frontend/src/api/invest.ts

```diff
@@ -0,0 +1,55 @@
+import { BigNumber } from 'bignumber.js';
+
+import API, { LoanState } from './index';
+import { LoanToInvestT } from '@/store/invest/types';
+import { getTokenSymbolsFromBlockchain, getAmountsWantedFromBlockchain, getAmountsGatheredFromBlockchain } from './utils';
+import { getSingleAmountWantedFromBlockchain, getSingleAmountGatheredFromBlockchain, getSingleTokenSymbolFromBlockchain } from './utils';
+
+
+export async function getLoansToInvest(cb) {
+  const api = await API.instance();
+  let currentUser = await api.currentUser();
+  let blockchainLoans = await api.loansByState(LoanState.Fundraising);
+  await Promise.all(blockchainLoans.map(loan => loan.updateStateFromBlockchain()));
+
+  const amountsWanted: BigNumber[] = await getAmountsWantedFromBlockchain(blockchainLoans);
+  const amountsGathered: BigNumber[] = await getAmountsGatheredFromBlockchain(blockchainLoans);
+  const loanTokenSymbols: string[] = await getTokenSymbolsFromBlockchain(blockchainLoans);
+
+  let loansToInvest: LoanToInvestT[] =
+    blockchainLoans.map(({
+      shortId,
+      owner,
+      parameters,
+  }, index): LoanToInvestT => ({
+        id: shortId,
+        userName: owner.ascii,
+        interestPermil: parameters.interestPermil,
+        fundraisingDeadline: parameters.fundraisingDeadline.format('LL'),
+        amountGathered: amountsGathered[index],
+        amountWanted: amountsWanted[index],
+        tokenSymbol: loanTokenSymbols[index]
+      }));
+  cb(loansToInvest);
+}
+
+export async function getLoanToInvest(shortId) {
+  const api = await API.instance();
+  let currentUser = await api.currentUser();
+  let blockchainLoan = await api.loan(shortId);
+  await blockchainLoan.updateStateFromBlockchain();
+
+  const amountWanted: BigNumber = await getSingleAmountWantedFromBlockchain(blockchainLoan);
```

Comment:
It seems like we always download those three at the same time - maybe add a `get{Single/Many}LoanFromBlockchain` that calls all three in parallel via `Promise.all`?

Reply:
I think that concatenation of lists: ```amountWanted```, ```amountsGathered``` and ```tokenSymbols``` into one list and than doing ```Promise.all``` will of course be a little bit faster, but also very hard to understand